src/clr-core/common/utils/aria-live.service.ts

```diff
@@ -0,0 +1,124 @@
+   * The reason for that is that we don't want to do additional work if the SR will ignore it.
+   *
+   * @param message - This could be simple string or HTMLElement
+   * @param politeness - 'polite', 'assertive' or 'off'
+   */
+  announce(message: string | HTMLElement, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): void {
+    if (!message || politeness === AriaLivePoliteness.off) {
+      return;
+    }
+
+    if (typeof message !== 'string' && this.isBrowser() && message.textContent) {
+      message = message.textContent;
+    }
+
+    // when there is no message do NOTHING!
+    if (!message) {
+      return;
+    }
+
+    if (!this.ariaLiveElement && this.isBrowser()) {
+      this.ariaLiveElement = this.createContainer();
+      this.ariaLiveElement.setAttribute('aria-live', politeness);
+      this.addTextToAriaLiveElement(message);
+    }
+  }
+
+  cleanup() {
+    if (this.isBrowser() && this.ariaLiveElement) {
+      this.document.body.removeChild(this.ariaLiveElement);
+      this.ariaLiveElement = null;
+    }
+  }
+
+  protected addTextToAriaLiveElement(message: string | HTMLElement) {
+    // This clearTimeout will stop all older messages from announcing
+    // in the case where the messages are comming too fast we gonna try to append only
+    // the last one. That's what the SR will try to do anyway.
+    clearTimeout(this.previousTimeout);
+
+    // This 100ms timeout is necessary for some browser + screen-reader combinations:
+    // - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
+    // - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
+    //   second time without clearing and then using a non-zero delay.
+    // (using JAWS 17 at time of this writing).
+    this.previousTimeout = setTimeout(() => {
+      if (this.ariaLiveElement) {
+        this.ariaLiveElement.textContent = <string>message;
+      }
+    }, ARIA_LIVE_TICK);
+  }
+
+  protected isBrowser() {
+    return typeof window !== 'undefined';
+  }
+
+  /**
+   * Create AriaLive DOM element as a last child of the document.
+   * After the element is created, we gonna apply Clarity class to hide it from
+   * the screen and set the `aria-live` politness.
+   *
+   * `clr-sr-only` is the CSS class that is used to hide the element from the screen.
+   *
+   * @remark
+   * Calling this method multiple times will create multiple DOM Elements, that
+   * won't be tracked and will be GC after the service is destroyed.
+   *
+   * @return AriaLive container as HTMLElement
+   *
+   */
+  protected createContainer(): HTMLElement {
+    const ariaLiveElement = this.document.createElement('div');
+
+    ariaLiveElement.setAttribute('id', this.id);
+    // Use clarity screen reader class to hide the dom element
+    // and fix the scrollbar shake
+    ariaLiveElement.classList.add('clr-sr-only');
+
+    ariaLiveElement.setAttribute('aria-atomic', 'true');
```

Comment:
Seems like we have an opportunity to introduce a bulk attribute setter. something along the lines of...

```
setAttributes:: Element, Array(Array(attribute_name, value, optional_if_fn)) -> void
```